### PR TITLE
Remove special everest config handling

### LINKED ISSUE #4335: Make Everest use config as-is, not using own concepts
Everest have some special treatment today in the config-code, this should be removed, and we should utilize what is already there. 

Blocked by #4401

### PR DESCRIPTION
**Pull request by valentin-krasontovitsch**
**Issue**
Resolves #4335

**Approach**
See commit message(s)

NB: This change, together with the respective changes in everest, should be transparent to the end users, hence this change can be skipped in release notes.

## Pre review checklist

- [X] Added appropriate release note label
- [X] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

**Comment by valentin-krasontovitsch** [comment]
it is expected that this would fail against everest at this time, as we have not merged the necessary changes into everest yet.

**Comment by eivindjahren** [comment]
It also fails against everest test. Will we not merge the changes in everest first?

**Comment by valentin-krasontovitsch** [comment]
not sure what will be merged first, or when - perhaps we should click the two buttons simultaneously? morten says we should merge everest first though

**Comment by valentin-krasontovitsch** [comment]
we're working on that PR right now, reviewing the review commits and waiting for the jenkins test run to show up and succeed

**Comment by valentin-krasontovitsch** [comment]
that is a very good point : )

File: src/ert/_c_wrappers/enkf/res_config.py
Comment (andreas-el) [inline src/ert/_c_wrappers/enkf/res_config.py:77]: Was the intent to test `config_dict_given` instead of `config_dict` ? The end result is still the same though.

**Review by andreas-el — APPROVED** [review]
Looks good to me! :rocket:
The everest egg-test has to be fixed before merging though.
Also left small comment regarding variable, -- but that is nitpick.
Well done!

File: src/ert/_c_wrappers/enkf/res_config.py
Comment (andreas-el) [inline src/ert/_c_wrappers/enkf/res_config.py:224]: These comments seem outdated?

File: src/ert/_c_wrappers/enkf/res_config.py
Comment (andreas-el) [inline src/ert/_c_wrappers/enkf/res_config.py:217]: These comments seem outdated?

**Review by eivindjahren — APPROVED** [review]
big- :brain:

**Comment by valentin-krasontovitsch** [comment]
jenkins, please test everest

**Comment by valentin-krasontovitsch** [comment]
jenkins, retest everest please

**Comment by valentin-krasontovitsch** [comment]
jenkins retest everest

**Pull request merged**
